src/sys/ptrace.rs

```diff
@@ -140,3 +150,110 @@ pub fn ptrace_setsiginfo(pid: Pid, sig: &siginfo_t) -> Result<()> {
         Err(e) => Err(e),
     }
 }
+
+//-------------------------- Second part: a high-level wrapper for ptrace ----------------------//
+
+#[cfg(target_arch = "x86_64")]
```

Comment:
This is why your tests are failing, you need to either limit `peekuser()` to x86_64, since that's the only play this register is defined for, or expand `Register` to more platforms.